www/lendingbot.js

```diff
@@ -24,25 +24,296 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
+           mincoins[coinobject] = coinamountobject;
+           localStorage.setItem("MinCoins",JSON.stringify(mincoins));
+        };
+    }   
+        
+        var timestring = shorttimestring(((data.log[i]).match(regexarray[0]))[1]);
+        localStorage.setItem('last_updated',((data.log[i]).match(regexarray[0]))[1]);
+        
+        var message = (data.log[i]).replace(((data.log[i]).match(regexarray[0]))[1],"");
+
+        if (data.log[i].indexOf("Error") >= 0) {
+            if (localStorage.getItem('notificationOnError') == "true"){notification(message,"warning");}
+            message = '<span data-type="table-danger">' + message + '</span>';
+            errornumber++;
+            if($('.btn-danger').hasClass('disabled'))  {} else {logdataSet.push(["Error",timestring,message.replace("Error: ","")]);}
+            
+        }
+        else if (data.log[i].indexOf("Placing") >= 0) {
+            if (localStorage.getItem('notificationOnLoan') == "true"){notification(message,"success");}
+            message = '<span data-type="table-success">' +  coinicon(message) + '</span>';
+            successnumber++;
+            
+            var resultarray = [];
+            $.each(regexarray,function(index,value){
+                resultarray.push(((data.log[i]).match(regexarray[index])[1]));
+            });
+            
+            dataSet.push([shorttimestring(resultarray[0]),coinicon(resultarray[1]),resultarray[2],resultarray[3],resultarray[4]]);
+            
+            if($('.btn-success').hasClass('disabled')) {} else {logdataSet.push(["Success",timestring,message]);}
+            
+        }
+        else if (data.log[i].indexOf("Canceling") >= 0) {
+            message = '<span data-type="table-info">' + message + '</span>';
+            if($('.btn-info').hasClass('disabled'))  {} else {logdataSet.push(["Info",timestring,message]);}
+            cancelednumber++;
+            
+        }
+        else {
+            message = '<span data-type="table-custom">' +  coinicon(message) + '</span>';
+            if($('.infobutton').hasClass('disabled'))  {} else {logdataSet.push(["Info",timestring,message]);}
+        }
+        
+        
+        $('#successnumber').text(successnumber);
+        $('#cancelednumber').text(cancelednumber);
+        $('#errornumber').text(errornumber);
+        $('#infonumber').text(data.log.length - (successnumber + cancelednumber + errornumber));
+       
     }
 
     updateOutputCurrency(data.outputCurrency);
     updateRawValues(data.raw_data);
+    
+    if ( $.fn.dataTable.isDataTable( '#openloans' ) ) {
+       mydatatable = $('#openloans').dataTable();
+    }
+    else {
+       mydatatable = $('#openloans').dataTable({responsive: true,createdRow: function( row, data, dataIndex ) {
+          $( row ).find('tr').addClass($( row ).find('td:eq(2) span').attr('data-type'));
+        }});
+    }
+
+    mydatatable.fnClearTable();
+    mydatatable.fnAddData(dataSet);
```

Comment:
what if "dataSet" is empty?
we can incur in "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error